### PR TITLE
TL-26984: 1PlusX boilerplate - ALT

### PR DESCRIPTION
**Pull request by nllerandi3lift**
Epic: https://triplelift.atlassian.net/browse/TL-27574
PR Review ticket: https://triplelift.atlassian.net/browse/TL-28208

This PR allows for our bid adapter to pick up the 1plusx localStorage and add to the bid request. The PR includes 2 tests: one to check that the relevant nested objects are created if no fpd exists; and the other that checks that the 1plusx segment is added to existing segments. 

- [x] !isEmpty(oneplusx) is prob overkill
- [x] what if fpdUser.data exists and it isn't an array (shouldn't be, but still)
- [x] check for malformed json following JSON.parse(oneplusx)

### CODE REVIEW

File: modules/tripleliftBidAdapter.js [hunk 1]
```diff
@@ -197,10 +199,26 @@ function _getGlobalFpd() {
...
 
+  // NEEDS TEST: check that oneplusx is parseable
+  if (oneplusx && _testJSONParse(oneplusx)) {
+    fpdUser.data = fpdUser.data || []
+    // NEEDS TEST: check that fpdUser.data is an array before pushing;
+    //    publisher may accidentally make it an object
+    try {
+      fpdUser.data.push({
+        name: '1PlusX',
+        segment: [oneplusx]
```
Comment (nllerandi3lift) [inline modules/tripleliftBidAdapter.js:215]: would need to JSON.parse(oneplusx) again here. think of better way to test and utilized parsed json in one shot

Comment (patrickloughrey) [reply modules/tripleliftBidAdapter.js:215]: pushed changes that adds a boolean flag which I think is a cleaner approach. It addresses this redundancy concern

Comment (nllerandi3lift) [reply modules/tripleliftBidAdapter.js]: I like where that went. Thanks. I just made a minor change. Check it out.

File: modules/tripleliftBidAdapter.js [hunk 2]
```diff
@@ -197,10 +199,26 @@ function _getGlobalFpd() {
   const context = {}
   const user = {};
   const ortbData = config.getConfig('ortb2') || {};
+  const onePlusXStorage = _fetchOnePlusX();
```
Comment (patrickloughrey) [inline modules/tripleliftBidAdapter.js:202]: I like this naming convention much better

File: modules/tripleliftBidAdapter.js [hunk 3]
```diff
@@ -10,6 +11,7 @@ const BANNER_TIME_TO_LIVE = 300;
 const INSTREAM_TIME_TO_LIVE = 3600;
 let gdprApplies = true;
 let consentString = null;
+export const storage = getStorageManager({gvlid: GVLID, bidderCode: BIDDER_CODE});
```
Comment (nllerandi3lift) [inline modules/tripleliftBidAdapter.js:14]: export const so `storage` can be used in test file

File: modules/tripleliftBidAdapter.js [hunk 4]
```diff
@@ -213,6 +228,18 @@ function _getGlobalFpd() {
   return fpd;
 }
 
+function _fetchOpeCloud() {
+  const opeCloud = storage.getDataFromLocalStorage('opecloud_ctx');
```
Comment (nllerandi3lift) [inline modules/tripleliftBidAdapter.js:232]: must use Prebid's storageManager to fetch local storage

**Comment by nllerandi3lift** [comment]
Added to connections-extend 88 for a consultative look

**Comment by nllerandi3lift** [comment]
Converting to draft as to not interfere with https://github.com/prebid/Prebid.js/pull/8488
Will merge and submit PR once the above is through

**Review by kzhouTL — APPROVED** [review]

**Pull request merged**